controllers/update_status.go

```diff
@@ -360,14 +362,18 @@ func (s UpdateStatus) Reconcile(r *FoundationDBClusterReconciler, context ctx.Co
 	sort.SliceStable(status.ProcessGroups, func(i, j int) bool {
 		return status.ProcessGroups[i].ProcessGroupID < status.ProcessGroups[j].ProcessGroupID
 	})
+
 	cluster.Status = status
 
 	_, err = cluster.CheckReconciliation()
 	if err != nil {
 		return false, err
 	}
 
-	if !reflect.DeepEqual(cluster.Status, *originalStatus) {
+	// See: https://github.com/kubernetes-sigs/kubebuilder/issues/592
+	// If we use the default reflect.DeepEqual method it will be recreating the
+	// status multiple times because the pointers are different.
+	if !equality.Semantic.DeepEqual(cluster.Status, *originalStatus) {
```

Comment:
@brownleej I changed this to be robust against different pointers but the same content `reflect.DeepEqual` will check if the pointers are the same and `equality.Semantic.DeepEqual` will check if the content is the same which makes more sense in our case.